### ISSUE TITLE
Expand Hide or Collapsing Text

### ISSUE THREAD

**Issue by ebonsi** [Minds/minds#28]
Sure, there are many ways to Expand and Collapse text. I am looking into these codes and I am wondering if it would be possible to incorporate into the channel timeline posting area. The code example can be found here; Thanks!
http://www.randomsnippets.com/2008/02/12/how-to-hide-and-show-your-div/

div id="headerDivImg">

    <div id="titleTextImg">Let's use images!</div>

    <a id="imageDivLink" href="javascript:toggle5('contentDivImg', 'imageDivLink');"><img src="/wp-includes/images/minus.png"></a>

</div>

<div id="contentDivImg" style="display: block;">This demo uses plus and minus images for hiding and showing your div dynamically via JavaScript.</div>

Here is the JavaScript code:

    function toggle5(showHideDiv, switchImgTag) {

            var ele = document.getElementById(showHideDiv);

            var imageEle = document.getElementById(switchImgTag);

            if(ele.style.display == "block") {

                    ele.style.display = "none";

    		imageEle.innerHTML = '<img src="/wp-includes/images/plus.png">';

            }

            else {

                    ele.style.display = "block";

                    imageEle.innerHTML = '<img src="/wp-includes/images/minus.png">';

            }

    }

I found another interesting approach that use on CSS and HTML,
http://adrianroselli.com/2015/07/showhide-script-free-which-means-css-only.html

HTML

<main>
  <ul>
    <li><a href="#Expando1">The Tick</a>
      <ul class="expando" id="Expando1">
        <li>Has superhuman strength and mass;</li>
        <li>Is nigh-invulnerable; </li>
        <li>Has powers that increase as the situation becomes more dramatic;</li>
        <li>Does not seem to require oxygen;</li>
        <li>Cannot keep his balance if his antennae are removed.</li>
      </ul>    
    </li>
  </ul>

  <ul>
    <li><a href="#Expando2">Sewer Urchin</a>
      <ul class="expando" id="Expando2">
        <li>Has slime &amp; stench secreting spikes on his suit enable him to stick on any surface;</li>
        <li>Is equipped with lemon grenades, butter shooters, bars of soap and other various apparel to aid him in his underground endeavors; </li>
        <li>Has an oxygen tank and mask which enable him to breath in the thickest of sewer sludge;</li>
        <li>Has a submarine that he garbage-picked.</li>
      </ul>    
    </li>
  </ul>

  <ul>
    <li><a href="#Expando3">Arthur</a>
      <ul class="expando" id="Expando3">
        <li>Has a suit that enables wobbly flight;</li>
        <li>Has an accounting background.</li>
      </ul>    
    </li>
  </ul>
</main>
<a href="#0" id="rerun-button" class="action-button rerun-button" onclick="Embed.refreshResultIFrame(); return false;">
        Rerun
      </a>

CSS
/* Standard Bits */

@import url(http://fonts.googleapis.com/css?family=Lato:400,700);

body {
  background-color: #6D695C;
  background-image: url(data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAMAAAAoLQ9TAAAACVBMVEUAAAAAAAAAAACDY+nAAAAAA3RSTlMmDQBzGIDBAAAAG0lEQVR42uXIIQEAAADCMHj/0NdkQMws0HEeAqvwAUGJthrXAAAAAElFTkSuQmCC);
  font-size: 100%;
  color: #fff;
  font-family: Lato, Arial, sans-serif;
  padding: 0;
  margin: 0;
  line-height: 1.4;
}

main {
  display: block;
  box-sizing: border-box;
  width: 90%;
  margin: 1em auto;
  padding: 1em 2em;
  color: #000;
  background-color: rgba(204,204,204,.7);
  border: .07em solid rgba(0,0,0,.5);
  border-radius: .5em;
}

/* Expando Bits */

.expando {
  display: none;
}

.expando:target {
  display: block;
}

**Comment by markharding** [comment]
Thanks for opening the issue. 

I think the best way to implement this would be to create an angular component which makes use of TemplateRef and have a simple `[class.m-collapsable-collapsed]="isCollapsed` variabe applied to it. 

Are you familiar with Angular 2?

**Comment by ebonsi** [comment]
Hi Mark, I know now that you mentioned... a combination of these elements on a web development;

    Custom Elements - Enables the extension of HTML through custom tags.
    HTML Imports - Enables packaging of various resources (HTML, CSS, JS, etc.).
    Template Element - Enables the inclusion of inert HTML in a document.
    Shadow DOM - Enables encapsulation of DOM and CSS.

By hey, thanks for bringing that up, got me be going in a reading chasing! :)

**Comment by ebonsi** [comment]
<!doctype html>
    <html ng-app>
      <head>
        <script src="https://ajax.googleapis.com/ajax/libs/angularjs/1.6.2/angular.min.js"></script>
      </head>
      <body>
        <div>
          <label>Name:</label>
          <input type="text" ng-model="yourName" placeholder="Enter a name here">
          <hr>
          <h1>Hello {{yourName}}!</h1>
        </div>
      </body>
    </html>

**Comment by markharding** [comment]
Merged with minds/front#23
